### PR TITLE
Updated README for ROS2 Gem

### PR DESCRIPTION
**Pull request by adamdbrw**
This PR corrects the README of the ROS2 Gem to point to its new repository. 
- links in ROS2 Gem instructions
- changed the example project to the vacuum sample

Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

### CODE REVIEW

File: Gems/ROS2/README.md [hunk 1]
```diff
@@ -66,17 +66,18 @@ before building.
...
 
-To use this Gem in your project, you need to register the Gem with O3DE. First, clone this repository. Then,
+To use this Gem in your project, you need to register the Gem with O3DE. First, clone the o3de-extras repository. Then,
 in o3de folder:
+
 ```
-scripts/o3de.sh register --gem-path <PATH_TO_CLONED_ROS2_GEM>
+scripts/o3de.sh register --gem-path <PATH_TO_ROS2_GEM>
 scripts/o3de.sh enable-gem -gn ROS2 -pp <PATH_TO_YOUR_PROJECT>
 ```
```
Comment (michalpelka) [inline Gems/ROS2/README.md:76]:
```suggestion
Assume that you cloned `o3de-extras` to `/home/foo/o3de-extras/` and your project lives in `/home/foo/my-awesome-project`, the registration looks like:
```
scripts/o3de.sh register --gem-path /home/foo/o3de-extras/Gems/ROS2
scripts/o3de.sh enable-gem -gn ROS2 -pp /home/foo/my-awesome-project
```

**Review by hultonha — APPROVED** [review]

**Comment by spham-amzn** [comment]
Looks like https://github.com/o3de/o3de-extras/pull/99 already fixed the ROS2 gem readme, is this PR still needed?

**Comment by adamdbrw** [comment]
@spham-amzn partially, yes, but not quite, as the README contains link to obsolete Warehouse Demo which I proposed to replace with https://github.com/o3de/RobotVacuumSample.

**Comment by adamdbrw** [comment]
@spham-amzn @hultonha rebased and now it only replaces the link to an obsolete project with Robot Vacuum.

File: Gems/ROS2/README.md (hunk 1 above)
Comment (adamdbrw) [reply Gems/ROS2/README.md:76]: https://github.com/o3de/o3de-extras/issues/111

**Review by michalpelka — APPROVED** [review]

**Comment by hultonha** [comment]
LGTM, just started a build now - https://jenkins.build.o3de.org/view/Projects/job/o3de-extras/view/change-requests/job/PR-96/1/ (hopefully soon with TIAF (tagging @jonawals) these types of changes won't need to run any tests as we know the change is just to a text file that can't break code 🥳 - we should think about adding TIAF to o3de-extras before too long @jonawals)

**Review by moraaar — APPROVED** [review]

**Review by spham-amzn — APPROVED** [review]

**Review by michalpelka — APPROVED** [review]

**Pull request merged**